pkg/resource/resource.go

```diff
@@ -360,7 +360,10 @@ func GetExternalTags(mg Managed) map[string]string {
 	if mg.GetClassReference() != nil {
 		tags[ExternalResourceTagKeyClass] = mg.GetClassReference().Name
 	}
-	if mg.GetProviderReference().Name != "" {
+	switch {
+	case mg.GetProviderConfigReference() != nil && mg.GetProviderConfigReference().Name != "":
+		tags[ExternalResourceTagKeyProvider] = mg.GetProviderConfigReference().Name
+	case mg.GetProviderReference() != nil && mg.GetProviderReference().Name != "":
```

Comment:
nitpick: would you mind adding a note or todo here to remove this branch once `Provider` type is removed?